core/trino-main/src/test/java/io/trino/sql/TestLikeFunctions.java

```diff
@@ -69,8 +80,55 @@ public void testLikeChar()
         assertFalse(likeChar(7L, utf8Slice("foob"), regex));
         assertFalse(likeChar(7L, offsetHeapSlice("foob"), regex));
 
-        assertFunction("cast('foob' as char(6)) LIKE 'f%b__'", BOOLEAN, true);
-        assertFunction("cast('foob' as char(7)) LIKE 'f%b__'", BOOLEAN, false);
+        // pattern shorter than value length
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo  '", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'fo_'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'fo%'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE '%foo'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE '_oo'", BOOLEAN, false);
+        assertFunction("CAST('foob' AS char(6)) LIKE 'f%b__'", BOOLEAN, true);
+        assertFunction("CAST('foob' AS char(7)) LIKE 'f%b__'", BOOLEAN, false);
+
+        // pattern of length equal to value length
+        assertFunction("CAST('foo' AS char(3)) LIKE 'foo'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(3)) LIKE 'fob'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo   '", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo __'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE '%%%%%%'", BOOLEAN, true);
+
+        // pattern longer than value length
+        assertFunction("CAST('foo' AS char(3)) LIKE '%%foo'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(3)) LIKE 'f\\_\\_' ESCAPE '\\'", BOOLEAN, false);
+        assertFunction("CAST('f__' AS char(3)) LIKE 'f\\_\\_' ESCAPE '\\'", BOOLEAN, true);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo    '", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo __ '", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE '_______'", BOOLEAN, false);
```

Comment:
Maybe make the pattern longer so that it's easier to see that it's too long.

Reply:
there is `// pattern longer than value length` above this group

also, i want to test "as close to equal as possible"